xbmc/pvr/dialogs/GUIDialogPVRChannelManager.cpp

```diff
@@ -872,20 +925,22 @@ void CGUIDialogPVRChannelManager::RenameChannel(const CFileItemPtr& pItem)
   }
 }
 
-bool CGUIDialogPVRChannelManager::PersistChannel(const CFileItemPtr& pItem, const std::shared_ptr<CPVRChannelGroup>& group, unsigned int* iChannelNumber)
+bool CGUIDialogPVRChannelManager::PersistChannel(const CFileItemPtr& pItem,
+                                                 const std::shared_ptr<CPVRChannelGroup>& group)
 {
   if (!pItem || !group)
     return false;
 
-  return group->UpdateChannel(pItem->GetPVRChannelInfoTag()->StorageId(),
-                              pItem->GetProperty("Name").asString(),
-                              pItem->GetProperty("Icon").asString(),
-                              static_cast<int>(pItem->GetProperty("EPGSource").asInteger()),
-                              ++(*iChannelNumber),
-                              !pItem->GetProperty("ActiveChannel").asBoolean(), // hidden
-                              pItem->GetProperty("UseEPG").asBoolean(),
-                              pItem->GetProperty("ParentalLocked").asBoolean(),
-                              pItem->GetProperty("UserSetIcon").asBoolean());
+  return group->UpdateChannel(
+      pItem->GetPVRChannelInfoTag()->StorageId(),
+      pItem->GetProperty(PROPERTY_CHANNEL_NAME).asString(),
+      pItem->GetProperty(PROPERTY_CHANNEL_ICON).asString(),
+      static_cast<int>(pItem->GetProperty(PROPERTY_CHANNEL_EPG_SOURCE).asInteger()),
```

Comment:
is the cast needed since you‘re calling `.asInteger()` ?

Reply:
No idea, just took over the code piece that was there before. I thought there must be a reason for the cast and just moved on.

Reply:
The cast is there since 2012. :-) => https://github.com/xbmc/xbmc/commit/c354924cd061e880b5c6b26aa91193ad09804765#diff-0a4af33d0ff6fb47f4cefc7badc9424b6c2431e94e55c72f91791e53b8904181R784

Reply:
It's because that call on `CVariant` return `int64_t`.

Reply:
Yep, but I guess I will not change this in this PR.

Reply:
no need to change, was just scratching my head, thx for clarifying 👍